packages/connect/src/SmoldotProvider/SmoldotProvider.test.ts

```diff
@@ -204,78 +185,92 @@ test('subscribe', async t => {
+// test('subscribe', async done => {
+//   const responses = [
+//     '{"jsonrpc":"2.0","result":"SUBSCRIPTIONID","id":1}',
+//     '{"jsonrpc":"2.0","method":"state_test","params":{"result":{"dummy":"state"},"subscription":"SUBSCRIPTIONID"}}'
+//   ];
+//   const ms = mockSmoldot(respondWith(responses));
+//   const provider = new SmoldotProvider(EMPTY_CHAIN_SPEC, testDb(), ms);
+
+//   await provider.connect();
+
+//   expect.assertions(2);
+//   return new Promise<void>((resolve, reject) => {
+//     return provider.subscribe('state_test', 'test_subscribe', [],  (error: Error | null, result: any) => {
+//       if (error !== null) {
+//         done.fail(error.message);
+//         reject();
+//       }
+
+//       expect(result).toEqual({ dummy: "state" });
+//       provider.disconnect().then(() => resolve());
+//     }).then(reply => {
+//       expect(reply).toBe("SUBSCRIPTIONID");
+//     });
+//   });
+// });
+
+// test('subscribe copes with out of order responses', async done => {
+//   const responses = [
+//     '{"jsonrpc":"2.0","method":"state_test","params":{"result":{"dummy":"state"},"subscription":"SUBSCRIPTIONID"}}',
+//     '{"jsonrpc":"2.0","result":"SUBSCRIPTIONID","id":1}'
+//   ];
+//   const ms = mockSmoldot(respondWith(responses));
+//   const provider = new SmoldotProvider(EMPTY_CHAIN_SPEC, testDb(), ms);
+
+//   await provider.connect();
+
+//   expect.assertions(2);
+//   return new Promise<void>((resolve, reject) => {
+//     return provider.subscribe('state_test', 'test_subscribe', [],  (error: Error | null, result: any) => {
+//       if (error !== null) {
+//         done.fail(error.message);
+//         reject();
+//       }
+
+//       expect(result).toEqual({ dummy: "state" });
+//       provider.disconnect().then(() => {
+//         resolve();
+//       });
+//     }).then(reply => {
+//       expect(reply).toBe("SUBSCRIPTIONID");
+//     });
+//   });
+// });
+
+// test('converts british english method spelling to US', async done => {
+//   const responses = [
+//     '{"jsonrpc":"2.0","result":"SUBSCRIPTIONID","id":1}',
+//     '{"jsonrpc":"2.0","method":"chain_finalisedHead","params":{"result":{"dummy":"state"},"subscription":"SUBSCRIPTIONID"}}'
+//   ];
+//   const ms = mockSmoldot(respondWith(responses));
+//   const provider = new SmoldotProvider(EMPTY_CHAIN_SPEC, testDb(), ms);
+
+//   await provider.connect();
+
+//   expect.assertions(2);
+//   return new Promise<void>((resolve, reject) => {
+//     return provider.subscribe('chain_finalizedHead', 'chain_subscribeFinalizedHeads', [],  (error: Error | null, result: any) => {
+//       if (error !== null) {
+//         done.fail(error.message);
+//         reject();
+//       }
+
+//       expect(result).toEqual({ dummy: "state" });
+//       return provider.disconnect().then(() => resolve());
+//     }).then(reply => {
+//       expect(reply).toBe("SUBSCRIPTIONID");
+//     });
+//   });
+// });
```

Comment:
There are still several commented out tests.  I'm happy to help getting these to work if you want me too.

Reply:
I just fixed those 😄